helm/cluster-gcp/values.yaml

```diff
@@ -31,6 +31,8 @@ bastion:
   # subnet:
 
 controlPlane:
+  api:
```

Comment:
This is a general question, not specific to this PR, so it's not meant to block it.

In my experience with helm, I tend to face issues when using nested values like
```
api:
  allowListSubnets:
```
instead of plain values like
```
apiAllowListSubnets
```

Mostly issues related to accidentally nullifying other subfields. Do you folks have opinions on when to use one style or the other?

Reply:
Here my brain just decided it would be neater to have `bastion.allowListSubnets` and `controlPlane.api.allowListSubnets`. There's no other api values currently and we're pre-v1, so if you think it's safer to have the flat value `apiAllowListSubnets` I'd be happy to change it

Reply:
I would use flat values for **everything** that we have in the values file except for the `machineDeployments` list, converting the root fields to prefixes, i.e `bastion.enabled` would become `bastionEnabled`, etc.

But I don't know if I'm being crazy about it.